src/gluonnlp/data/utils.py

```diff
@@ -407,3 +406,35 @@ def __call__(self, s):
             List of strings. Obtained by calling s.split(separator).
         """
         return s.split(self._separator)
+
+
+def _convert_to_unicode(text):
+    """Converts `text` to Unicode.
+
+    Parameters
+    ----------
+    text : str or bytes
+        text to be converted to unicode
+
+    Returns
+    -------
+    str
+        unicode string
+    """
+    py_version = sys.version_info[0]
+    if py_version == 3:
```

Comment:
python version won't change during the process, so the version check can be moved outside the function so that the check only happens once.